### PR TITLE
Humble v1.5.0 Release

### PR DESCRIPTION
**Pull request by mhubii**
This PR targets `1.5.0`. This will likely be the last major `humble` release and should pave the way for ROS index binaries. Following PRs will aim at `rolling`.

- Targeted release **14/04/2024**
- Much improved integrability
- Much improved testing
- Better demos

# Contributors
- https://github.com/lbr-stack/fri/pull/22: fredRocs
- https://github.com/lbr-stack/fri/pull/23: peterMitrano
- https://github.com/lbr-stack/fri/pull/18: bowangFromMars
- FRI 2.x testing: liver121888
- #151: Nicolai-98 

# Related PRs
- #158 
- #155 
- #151 
- #161 
- #170

# Issues Tracker

- #9 (following `1.5.0`)
- #163 
- #165 
- #172
- #159

# Desired Changes
- [x] Black linting
- [x] Fix link to docs
- [x] Velocity limit checks in impedance control mode
- [x] Matrix testing against multiple FRIs
- [x] ~~Add `${ROS_DISTRO}` to install instruction~~ (would require source, which renders it redundant) 
- [x] ~~Introduce node composition in launch files~~ to be done later
  Blockers:
  - `ros2_control` via composition: https://github.com/ros-controls/ros2_control/issues/1261
  - `rviz2` via composition: https://github.com/ros2/rviz/issues/1114
- [x] Add development tools dependency https://github.com/lbr-stack/lbr_fri_ros2_stack/issues/145
- [x] `ros2_control_node`: Read robot description from robot state publisher
- [x] ~~Use an event handler for robot state publisher~~ https://github.com/ros-controls/ros2_control/issues/1262
- [x] Add tests
  - [x] `lbr_fri_msgs`
  - [x] `lbr_fri_ros2` interface tests
- [x] De-couple interfaces by client command mode
- [ ] Update launch files
  - [x] Remove redundant nones from launch
  - [ ] De-couple launch files for simplified custom setups 
    - [ ] Add separate launch files for real / sim as no feature parity (would require Gazebo plugins)
    - [ ] Fix force torque broadcaster rviz config for simulation
    - [ ] Make separate moveit launch default
  - [x] Consider moving all launch mixins to `lbr_bringup`, or into `launch_mixins`, see https://github.com/mhubii/demo_launch
  - [ ] Add configurable static transform at launch, i.e. `world` -> `lbr/world`
  - [x] Remove `app.launch.py` from demos in favor for `ros2_control` variant (⚠️ breaking change)
- [ ] Update documentation for `v1.4.x`
  - [ ] Refer https://lbr-fri-ros2-stack-doc.readthedocs.io/en/latest/index.html
  - [x] Delete branches at https://github.com/lbr-stack/lbr_stack_doc in favor of tags
  - [ ] Add an architecture chart to highlight relation to `ros2_control` in readme
- [x] Update parameter source for `gazebo_ros2_control` package in `lbr.gazebo.xacro`
- [x] Add coloring a la https://github.com/ros-controls/ros2_control/blob/e149646d3f2595f269cfa4e1cd0681abde89ee69/controller_manager/controller_manager/spawner.py#L45
- [x] De-couple the async client into position / torque / wrench
- [x] Update Docker for new sources
- [x] `lbr_fri_msgs` -> `lbr_fri_idl` (⚠️ breaking change)
- [ ] Build docs on PR: https://docs.readthedocs.io/en/stable/pull-requests.html

FvnPq61X0AEbjNd

https://x.com/jdorfman/status/1655582823082782721?s=20

### CODE REVIEW

File: lbr_demos/doc/lbr_demos.rst [hunk 1]
```diff
@@ -5,6 +5,9 @@ Demos for controlling the LBR through the Fast Robot Interface (FRI) from ROS 2.
 .. warning::
     On the real robot, do always execute in ``T1`` mode first.
 
+TODO: Differentiate demos through controllers
```
Comment (mhubii) [inline lbr_demos/doc/lbr_demos.rst]: update this doc

File: lbr_fri_ros2/doc/lbr_fri_ros2.rst [hunk 2]
```diff
@@ -1,5 +1,7 @@
 LBR FRI ROS 2
 =============
+TODO: massive re-write of this section, since architecture has changed.
```
Comment (mhubii) [inline lbr_fri_ros2/doc/lbr_fri_ros2.rst]: update this doc

File: lbr_bringup/doc/lbr_bringup.rst [hunk 3]
```diff
@@ -1,34 +1,7 @@
 LBR Bringup
 ===========
-The ``lbr_fri_ros2_stack`` is designed for research **and** deployment. It runs standalone, with ``ros2_control``, and thus also with ``MoveIt 2``. Details are described in below sections
+TODO: Mention different controllers here already
```
Comment (mhubii) [inline lbr_bringup/doc/lbr_bringup.rst]: update this doc

File: lbr_description/lbr_description/launch_mixin.py [hunk 4]
```diff
@@ -30,43 +30,46 @@ def include_gazebo(**kwargs) -> IncludeLaunchDescription:
 
     @staticmethod
     def node_spawn_entity(
-        robot_name: Optional[Union[LaunchConfiguration, str]] = None, **kwargs
+        robot_name: Optional[Union[LaunchConfiguration, str]] = LaunchConfiguration(
+            "robot_name", default="lbr"
+        ),
+        tf: List[float] = [0.0, 0.0, 0.0, 0.0, 0.0, 0.0],
```
Comment (mhubii) [inline lbr_description/lbr_description/launch_mixin.py:36]: this transform does not work with moveit out of the box

**Comment by shengyangzhuang** [comment]
Hey! v1.5.0 doesn't seem to be able to build due to the lbr_demos related packages. Related files involving package.xml, CMakeList.txt, setup.py, setup.cfg, /resource/~ need to be modified.

**Comment by mhubii** [comment]
hi @shengyangzhuang , yes sorry, this is a little under construction

**Comment by mhubii** [comment]
hi @shengyangzhuang this PR now lives on the `humble` branch. If you find time, please re-install drivers following

https://github.com/lbr-stack/lbr_fri_ros2_stack#quick-start

there was an issue with the previous version that can trigger an unexpected motion. This issue is now fully fixed.

**Pull request merged**